<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [ ]:
# Membuat dataframes dari dictionary
import pandas as pd

Data_Pekerja = {
    'Nama' : ['Bayu', 'Topan', 'Arga','Rhesa','Hani'],
    'Role' : ['staff', 'staff', 'intern', 'spv', 'intern'],
    'Umur' : [27, 25, 21, 30, 23],
    
}

df_pekerja = pd.DataFrame(Data_Pekerja)
df_pekerja.head()

Nama    Role  Umur
0   Bayu   staff    27
1  Topan   staff    25
2   Arga  intern    21
3  Rhesa     spv    30
4   Hani  intern    23

<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [ ]:
# upload data, karena file data diluar google colab maka harus upload dulu, kalau di jupyter notebook maka define pathnya semisal terhubung lokal
#contoh :
#import pandas as pd
#df = pd.read_csv (r'Path where the CSV file is stored\File name.csv')
#print (df)
from google.colab import files
upload = files.upload()


Saving diamonds-1.csv to diamonds-1.csv


In [ ]:
# baca dan assign ke suatu variabel
df_diamonds = pd.read_csv('diamonds-1.csv') #upload dulu file diamonds-1.csv ke File sebelum di read pada collab
df_diamonds

Unnamed: 0  carat        cut color  ... price     x     y     z
0               1   0.23      Ideal     E  ...   326  3.95  3.98  2.43
1               2   0.21    Premium     E  ...   326  3.89  3.84  2.31
2               3   0.23       Good     E  ...   327  4.05  4.07  2.31
3               4   0.29    Premium     I  ...   334  4.20  4.23  2.63
4               5   0.31       Good     J  ...   335  4.34  4.35  2.75
...           ...    ...        ...   ...  ...   ...   ...   ...   ...
53935       53936   0.72      Ideal     D  ...  2757  5.75  5.76  3.50
53936       53937   0.72       Good     D  ...  2757  5.69  5.75  3.61
53937       53938   0.70  Very Good     D  ...  2757  5.66  5.68  3.56
53938       53939   0.86    Premium     H  ...  2757  6.15  6.12  3.74
53939       53940   0.75      Ideal     D  ...  2757  5.83  5.87  3.64

[53940 rows x 11 columns]

Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawab: 



*   **Series pada pandas mempunyai index atau key yang dapat disesuaikan** sedangkan **list hanya mempunyai index saja** dan **dictionary hanya mempunyai key saja**
*   **Dataframe merupakan sekumpulan dari  data series** dimana objeknya 2 dimensi, punya index baris dan kolom sedangkan **data series hanya terdiri dari single list dengan index** dimana objeknya 1 dimensi



In [ ]:
# contoh data series
kota_RegJabar = pd.Series(['Jakarta', 'Bandung', 'Bogor', 'Tasikmalaya', 'Garut', 'Majenang'], index=['kota_1', 'kota_2', 'kota_3', 'kota_4', 'kota_5', 'kota_6']) 
kota_RegJabar

kota_1        Jakarta
kota_2        Bandung
kota_3          Bogor
kota_4    Tasikmalaya
kota_5          Garut
kota_6       Majenang
dtype: object

In [ ]:
print(type(kota_Jabar))

<class 'pandas.core.series.Series'>


<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [ ]:
# 5 baris pertama dari diamonds-1.csv untuk pengecekan
df_diamonds.head()

Unnamed: 0  carat      cut color clarity  ...  table  price     x     y     z
0           1   0.23    Ideal     E     SI2  ...   55.0    326  3.95  3.98  2.43
1           2   0.21  Premium     E     SI1  ...   61.0    326  3.89  3.84  2.31
2           3   0.23     Good     E     VS1  ...   65.0    327  4.05  4.07  2.31
3           4   0.29  Premium     I     VS2  ...   58.0    334  4.20  4.23  2.63
4           5   0.31     Good     J     SI2  ...   58.0    335  4.34  4.35  2.75

[5 rows x 11 columns]

In [ ]:
# Baris pertama dari column price
print(df_diamonds.iloc[0]['price'])

326


In [ ]:
#  5 baris terakhir dari diamonds-1.csv untuk pengecekan
df_diamonds.tail()

Unnamed: 0  carat        cut color  ... price     x     y     z
53935       53936   0.72      Ideal     D  ...  2757  5.75  5.76  3.50
53936       53937   0.72       Good     D  ...  2757  5.69  5.75  3.61
53937       53938   0.70  Very Good     D  ...  2757  5.66  5.68  3.56
53938       53939   0.86    Premium     H  ...  2757  6.15  6.12  3.74
53939       53940   0.75      Ideal     D  ...  2757  5.83  5.87  3.64

[5 rows x 11 columns]

In [ ]:
# Baris terakhir dari column color
print(df_diamonds.iloc[53939]['color']) #atau bisa juga iloc[-1], dibaca dari bawah dimulai dari -1 dan seterusnya, maksud -1,-2,-3,....
print(df_diamonds.tail().iloc[4]['color']) #jika dibaca dari tailnya saja

D
D


<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Jawab: 

**statistik deskriptif adalah ilmu yang menjelaskan tentang penyajian dan pengumpulan gugus data untuk diolah sehingga dapat ditarik informasi yang berguna**

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [ ]:
# deskriptif statistik
df_diamonds.describe()

Unnamed: 0         carat  ...             y             z
count  53940.000000  53940.000000  ...  53940.000000  53940.000000
mean   26970.500000      0.797940  ...      5.734526      3.538734
std    15571.281097      0.474011  ...      1.142135      0.705699
min        1.000000      0.200000  ...      0.000000      0.000000
25%    13485.750000      0.400000  ...      4.720000      2.910000
50%    26970.500000      0.700000  ...      5.710000      3.530000
75%    40455.250000      1.040000  ...      6.540000      4.040000
max    53940.000000      5.010000  ...     58.900000     31.800000

[8 rows x 8 columns]

In [ ]:
# nilai rata-rata dari column price
print(df_diamonds['price'].mean())

# nilai standar deviasi dari column depth
print(df_diamonds['depth'].std())

# nilai maximum dari column carat
print(df_diamonds['carat'].max())

3932.799721913237
1.4326213188337733
5.01


Jawab:



*   Metode pandas untuk mengetahui statistik deskriptif dari suatu data adalah :

    "nama variabel".describe()

*   Nilai rata-rata dari column price: 3932.7998

* Nilai standar deviasi dari column depth: 1.4326

* Nilai maksimum dari column carat: 5.01

